### ISSUE TITLE
Twitter integration discussion

### ISSUE THREAD

**Issue by aminamos** [golf1052/SeattleCarsInBikeLanes#11]
Creating this to discuss current Twitter happenings, and what that could mean for sign-in and other functionality. Specific items of concern:

- can't view links to tweets without being signed in
- potential for rate limiting, inability to read tweets

**Comment by golf1052** [comment]
Yeah I've been thinking about how to best move forward.

For the website I've already changed embeds to use Mastodon. This works for basically 99% of all submissions.

In terms of submissions I've pushed for people submitting on Twitter to use the website instead for submissions.

I'll need to do one more pass through the code base but I'm pretty sure doesn't depend on Twitter at all anymore and I could submit and accept everything on Mastodon if need be.

**Comment by eeshvardasikcm** [comment]
I don't understand. I haven't skimmed through the updates in a while. Are you talking about getting submissions off of Twitter entirely, instead of integrating?

**Comment by aminamos** [comment]
At the time, Twitter (now X) blocked non-signed-in users from seeing tweets. Signed in users were only able to read a set ceiling amount of tweets per day.

**Comment by golf1052** [comment]
TL;DR: If there is interest in fixing "Sign in with Twitter" functionality on the website please let me know here and I'll ask on the Twitter developer forums to get my account reviewed.

If a large majority of reports came in through the website directly rather than Twitter than yes I would prefer to not post on Twitter.

Currently I manually post every post to Twitter before using the admin panel to post to all other socials (Mastodon and Bluesky). This is because my Twitter application has been suspended since about April. It had been randomly suspended in the past but after a few days it would get automatically unsuspended. I assume this was due to a glitch on Twitter's end. The April suspension hasn't been lifted yet. I have followed what limited guidance there is on Twitter's developer pages but it seems like the only way forward is to request on the [Twitter community forums](https://twittercommunity.com/) for a Twitter staff member to review your account and lift the suspension. I cannot contact support directly to open a case with them because Twitter has removed a way for devs to open support cases directly. I haven't asked on the forums yet because according to posts from other devs their accounts are still getting randomly suspended again, even after getting fixed by Twitter support. I really do not feel like asking on the forums, continually checking and pinging the support account, to hope to get un-suspended only to get re-suspended in the future because of bugs on Twitter's end.

The workflow of uploading manually to Twitter and then using the admin page to post to other socials works adequately. It is much slower than just pressing upload and I have to be on desktop in order to upload, but it does work. The one upside is that now that Threads ([my account here](https://www.threads.net/@carbikelanesea)) has desktop support I could also start manually posting to Threads without much more additional friction. It would require code changes to store Threads links in the DB but that would be a very quick change.

Unfortunately because of the Twitter application being suspended "Sign in with Twitter" on the website was broken. I removed the option from the website on 2023-08-05 (https://github.com/golf1052/SeattleCarsInBikeLanes/commit/833d90e5c7a43d0dbb64a6f479588819efdaaf97). People that post reports or send me reports directly on Twitter can still be properly attributed through the DB and on other socials. It's just that the website no longer supports logging in with Twitter to attribute submissions to your Twitter account. People can attribute themselves using Mastodon on the website.

If there is interest in fixing "Sign in with Twitter" functionality on the website please let me know here and I'll ask on the Twitter developer forums to get my account reviewed. If not I'll close this issue in the next few weeks.
